src/topology/config/mod.rs

```diff
@@ -118,6 +121,32 @@ pub enum DataType {
     Metric,
 }
 
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+pub struct LogSchemaOptions {
+    #[serde(default = "LogSchemaOptions::default_host_key")]
+    pub host_key: Atom,
+    #[serde(default = "LogSchemaOptions::default_message_key")]
+    pub message_key: Atom,
+    #[serde(default = "LogSchemaOptions::default_timestamp_key")]
+    timestamp_key: Atom,
+}
```

Comment:
We've discussed reducing our `pub` exposure in the tech debt issue. Consider if you can reduce any of these to `pub(crate)` or eliminate the `pub` entirely to improve what the compiler is able to check for us.